api/v1/routes/submissions.rb

```diff
@@ -26,34 +26,20 @@ class Submissions < Core
           halt 404, { error: "unknown submission #{key}" }.to_json
         end
 
-        status, payload = Xapi.get("v2", "exercises", submission.track_id, submission.slug)
-        halt status, payload if status != 200
-
-        begin
-          data = JSON.parse(payload)
-        rescue
-          halt 500, { error: "cannot get problem #{submission.slug} in #{submission.track_id}" }.to_json
-        end
-
-        problems = data['problems']
-        if problems.nil?
-          halt 404, { error: "#{submission.slug} in #{submission.track_id} not found" }.to_json
-        end
-
-        exercise = problems.first
+        exercise = Trackler.tracks[submission.language].implementations[submission.slug]
         if exercise.nil?
           halt 404, { error: "#{submission.slug} in #{submission.track_id} not found" }.to_json
         end
 
         {
-          language: exercise['language'],
```

Comment:
> I'm not sure what you mean (because, clearly, this code is ridiculously confusing!).

It was a bit of a joke because it seems like they're the same value.

I think the only thing that we need to pull in from the submission is the user submitted code. I think this covers that.